base/cvd/cuttlefish/host/libs/web/android_build_api.cpp

```diff
@@ -359,7 +358,13 @@ Result<void> BuildApi::ArtifactToFile(const DeviceBuild& build,
                 << "Received \"" << json << "\"");
   CF_EXPECT(json.isMember("signedUrl"),
             "URL endpoint did not have json path: " << json);
-  std::string url = json["signedUrl"].asString();
+  return json["signedUrl"].asString();
+}
+
+Result<void> BuildApi::ArtifactToFile(const DeviceBuild& build,
+                                      const std::string& artifact,
+                                      const std::string& path) {
+  const auto url = CF_EXPECT(GetArtifactDownloadUrl(build, artifact));
   CF_EXPECT(CF_EXPECT(http_client->DownloadToFile(url, path)).HttpSuccess());
```

Comment:
not part of this CL, but nested `CF_EXPECT` calls will generate messy errors as the macro expansion of the inner `CF_EXPECT` will become part of the "failing expression" text of the outer `CF_EXPECT`.

Reply:
Created #539 to address this